lib/auth/auth_with_roles.go

```diff
@@ -1490,6 +1497,63 @@ func (a *ServerWithRoles) ListUnifiedResources(ctx context.Context, req *proto.L
 	}, nil
 }
 
+func (a *ServerWithRoles) filterICPermissionSets(r services.AccessCheckable, checker *resourceAccess) error {
+	app, isApp := r.(*types.AppV3)
+	if !isApp {
+		return trace.BadParameter("resource must be an app")
+	}
+
+	pss := app.Spec.IdentityCenter.GetPermissionSets()
+	if pss == nil {
+		return nil
+	}
+
+	assignment := services.IdentityCenterAccountAssignment{
+		AccountAssignment: &identitycenterv1.AccountAssignment{
+			Kind:    types.KindIdentityCenterAccountAssignment,
+			Version: types.V1,
+			Metadata: &headerv1.Metadata{
+				Name: "rbac-tester",
+			},
+			Spec: &identitycenterv1.AccountAssignmentSpec{
+				AccountId:     app.GetName(),
+				PermissionSet: &identitycenterv1.PermissionSetInfo{},
+			},
+		},
+	}
+	permissionSetQuery := assignment.Spec.PermissionSet
+	checkable := types.Resource153ToResourceWithLabels(assignment)
+
+	var requestableAccessChecker resourceAccessChecker
+	regularAccessChecker := checker.accessChecker
+	if checker.baseAuthChecker != nil {
+		regularAccessChecker = checker.baseAuthChecker
+		requestableAccessChecker = checker.accessChecker
+	}
```

Comment:
Can the logic from https://github.com/gravitational/teleport/blob/master/lib/auth/auth_with_roles.go#L1278  be reused. 

At first glance seems the to almost the same but instead of `c.requestableMap[resource.GetName()] = struct{}{}` this code sets the `ps.RequiresRequest = true`